### ISSUE TITLE
Repo2docker ignores other configuration files when Dockerfile present

### ISSUE THREAD

**Issue by tjcrone** [yuvipanda/hubploy#49]
This is more in the category of dumb question. Recently I learned that when a Dockerfile is present, [all other configuration files will be ignored](https://repo2docker.readthedocs.io/en/latest/config_files.html#config-files) by repo2docker. And indeed the command line version of repo2docker works like this. However Pangeo deployments all have a Dockerfile, and I cannot find any code in hubploy that would modify the behavior of r2d. So, how does this work? How can we include a Dockerfile and configuration files and make r2d work? Wondering because I'm trying to find a way to easily build Pangeo images locally and I don't think the CLI version of repo2docker will do it. Thanks @yuvipanda!

**Comment by tjcrone** [comment]
I think I figured it out. If we just run docker build ., then the onbuild image will run r2d_overlay.py, which will fold in the repo2docker stuff. Got it.

**Comment by yuvipanda** [comment]
Yeah, it's the r2d_overlay.py file, which needs to become its own project...

**Comment by tjcrone** [comment]
Are you sure? I feel like r2d_overlay.py makes things extra opaque, and solves a problem (using repo2docker with a base image) in a way that might better be solved another way. Do we need to use repo2docker in hubploy? Why don't we just use a Dockerfile? It seems like everything repo2docker does can be done in a Dockerfile. And it can come from a repo. Thoughts?

**Comment by yuvipanda** [comment]
@tjcrone if you put a Dockerfile in a hubploy image repo, repo2docker will simply pass through to Docker. So using the repo2docker specific files is an option for folks who don't wanna write a Dockerfile (of which there are many!).

r2d_overlay.py is a way to do extra customizations on top of a repo2docker built image if you do not wanna write a Dockerfile. It is currently only used by the PANGEO project.

Basically, if you want to and are capable of writing a Dockerile, you can ignore that repo2docker exists.

**Comment by yuvipanda** [comment]
Going to close this, per https://github.com/pangeo-data/pangeo-stacks/issues/125#issuecomment-584489689 as r2d_overlay.py has nothing to do with hubploy
